mos-platform/geos-cbm/crt0.c

```diff
@@ -0,0 +1,102 @@
+/***********************************************************************
+ * crt0.c
+ *
+ * This file contains the implemantation of the "C" definition of
+ * init and fini code and the directory entry and info sector for
+ * use in GEOS applications  with llvm-mos C compiler.
+ *
+ ***********************************************************************/
+
+#include "geos.h"
+
+asm(".set __zp_reg_size, 0xfb - 0x90 \n"
+    ".section .data \n"
+    "__userzp_copy:	.space __zp_reg_size+1 \n"
+    ".section .text  \n"
+    ".global __swap_userzp \n"
+    "__swap_userzp: \n"
+    "     php                             ; save all registers and flags \n"
+    "     pha \n"
+    "     txa \n"
+    "     pha \n"
+    "     tya \n"
+    "     pha \n"
+    "     ldx     # __zp_reg_size           ; $FA..$90 \n"
+    "__swaploop: \n"
+    "     ldy     __zp_reg_start,x        ; load zp byte \n"
+    "     lda     __userzp_copy,x         ; load buffer byte \n"
+    "     sta     __zp_reg_start,x        ; store zp byte \n"
+    "     tya \n"
+    "     sta     __userzp_copy,x         ; store buffer byte \n"
+    "     dex \n"
+    "     bpl     __swaploop \n"
+    "     pla                             ; restore all registers and flags \n"
+    "     tay \n"
+    "     pla \n"
+    "     tax \n"
+    "     pla \n"
+    "     plp \n"
+    "     rts \n");
+
+asm(".section .init.010,\"axR\",@progbits \n"
+    "jsr __swap_userzp \n");
+
+asm(".global __after_main\n"
+    ".section .after_main,\"axR\",@progbits\n"
+    "__after_main:\n"
+    "     jsr __swap_userzp\n"
+    "     jmp __EnterDeskTop\n");
+
+void EnterDeskTop(void) {
+  __attribute__((leaf)) asm volatile("jsr __swap_userzp \n"
+                                     "jmp __EnterDeskTop\n");
+}
+
+__attribute__((section(".dir_block")))
+__attribute__((retain))
+__attribute__((weak)) 
+cvt_dir_entry_t __std_dir_entry = {
+    {USR,
```

Comment:
Since this opens a struct, the contents should be indented. Did clang-format do that? If so; disregard; it's a losing battle fighting clang-format, unfortunately.

Reply:
It was screwed by clang-format, so i keep it this way